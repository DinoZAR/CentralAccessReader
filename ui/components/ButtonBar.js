import React from 'react';

import { StyleSheet, css } from '../styles/aphrodite';
import constants from '../styles/constants';

class ButtonBar extends React.Component {
  render() {
    return (
      <div className={css(styles.bar)}>
        <button className={css(styles.button)}>
          <i className="fa fa-play fa-3x" />
        </button>
        <button className={css(styles.button)}>
          <i className="fa fa-cog fa-3x" />
        </button>
        <button className={css(styles.button)}>
          <i className="fa fa-music fa-3x" />
        </button>
        <div className={css(styles.smallSpacer)} />
        <button className={css(styles.button)}>
          <i className="fa fa-search-plus fa-3x" />
        </button>
        <button className={css(styles.button)}>
          <i className="fa fa-search-minus fa-3x" />
        </button>
        <button className={css(styles.button)}>
          <i className="fa fa-search fa-3x" />
        </button>
        <div className={css(styles.restSpacer)} />
        <button className={css(styles.button)}>
          <i className="fa fa-columns fa-3x" />
        </button>
      </div>
    );
  }
}

const styles = StyleSheet.create({
  bar: {
    backgroundColor: constants.lighterBackgroundColor,
    display: 'flex',
    flexDirection: 'column',
    width: '5rem',
    height: '100%',
    boxShadow: constants.panelShadow,
  },
  button: {
    height: '4rem',
    backgroundColor: constants.lighterBackgroundColor,
    color: 'white',
    border: 'none',
    ':hover': {
      backgroundColor: constants.primaryColor,
      cursor: 'pointer',
    }
  },
  smallSpacer: {
    flexGrow: 1,
    maxHeight: '4rem'
  },
  restSpacer: {
    flexGrow: 1,
  }
});

export default ButtonBar;