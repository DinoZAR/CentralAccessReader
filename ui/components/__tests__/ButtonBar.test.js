import React from 'react';
import ButtonBar from '../ButtonBar';

describe('ButtonBar', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<ButtonBar />);
    expect(wrapper).toMatchSnapshot();
  });
});
