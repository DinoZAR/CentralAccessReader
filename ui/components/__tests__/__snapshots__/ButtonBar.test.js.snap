// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonBar renders correctly 1`] = `
<div>
  <button>
    <i
      className="fa fa-play fa-3x"
    />
  </button>
  <button>
    <i
      className="fa fa-cog fa-3x"
    />
  </button>
  <button>
    <i
      className="fa fa-music fa-3x"
    />
  </button>
  <div />
  <button>
    <i
      className="fa fa-search-plus fa-3x"
    />
  </button>
  <button>
    <i
      className="fa fa-search-minus fa-3x"
    />
  </button>
  <button>
    <i
      className="fa fa-search fa-3x"
    />
  </button>
  <div />
  <button>
    <i
      className="fa fa-columns fa-3x"
    />
  </button>
</div>
`;
